file.js

```diff
@@ -7,14 +7,55 @@ module.exports = function (file, block_size, flags) {
   flags = flags || 'r+'
   var fd
   var offset = Obv()
-  var writing = false
-  var waitingForWrite = []
-
-  function readyToWrite () {
-    if(!writing) throw new Error('should be writing')
-    writing = false
-    while(waitingForWrite.length)
-      waitingForWrite.shift()()
```

Comment:
Just to clarify, this was the part that wasn't async. I think the behavior was supposed to be:

- when a non-queued function finishes they call `readyToWrite()`
- this function checks the queue and runs each function
  - functions should be run one at a time
  - when these functions finish they should die

What ended up happening:

- :heavy_check_mark:  when a non-queued function finishes they call `readyToWrite()`
- :x:  this function checks the queue and runs each function
  - :scream:  async functions return immediately so they all run at the same time
  - :bug: these functions call `readyToWrite()` again and start consuming the queue as well
    - :bomb: this ends up looking like a fork bomb where each function spawns more functions to eat the queue